src/dataset/infrastructure/repositories/DatasetJSDataverseRepository.ts

```diff
@@ -200,23 +200,36 @@ export class DatasetJSDataverseRepository implements DatasetRepository {
       getDatasetSummaryFieldNames.execute(),
       getPrivateUrlDatasetCitation.execute(privateUrlToken)
     ])
-      .then(([jsDataset, summaryFieldsNames, citation]: [JSDataset, string[], string]) =>
-        JSDatasetMapper.toDataset(
+      .then(async ([jsDataset, summaryFieldsNames, citation]: [JSDataset, string[], string]) => {
+        const [permissions, locks, originalSize, archivalSize] = await Promise.all([
+          getDatasetUserPermissions.execute(jsDataset.id),
+          getDatasetLocks.execute(jsDataset.id),
+          getDatasetFilesTotalDownloadSize.execute(
+            jsDataset.id,
+            DatasetNonNumericVersion.DRAFT,
```

Comment:
@ChengShi-1 Looks great, just a question, maybe a silly one, why pass the DRAFT version to the `getDatasetFilesTotalDownloadSize` and not make it dynamic according to the version of the dataset in the view?

Reply:
Hi German, thanks so much for reviewing! I should add the dynamic input "version" to the function. I skipped it because I mistakenly assumed the result would be the same regardless of the version.

Reply:
I think so, maybe we could ask @ekraffmiller about this?

Reply:
Thanks! Approving then 👍🏼